tests/acceptance/blocking_adapter_test.py

```diff
@@ -476,13 +469,13 @@ def test(self):
         self.assertIsInstance(method, pika.spec.Basic.GetOk)
         self.assertEqual(method.delivery_tag, 1)
         self.assertFalse(method.redelivered)
-        self.assertEqual(method.exchange, _p3_as_bytes(''))
-        self.assertEqual(method.routing_key, q_name)
+        self.assertEqual(method.exchange, as_bytes(''))
+        self.assertEqual(method.routing_key, as_bytes(q_name))
```

Comment:
Thanks for entertaining this thought. It should be much more robust to deal with all `shortstr` types as strings (or unicode strings). This would make equality tests work intuitively for those fields. (http://stackoverflow.com/questions/1864701/convert-utf-8-octets-to-unicode-code-points)

Only `longstr` should be bytes per AMQP definition that you sited. I am guessing that `body` is `longstr`, right?